jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -216,10 +224,22 @@ def _get_asn_id(self, input_models):
         self.log.info(f"Outlier Detection asn_id: {asn_id}")
         return
 
-    def _set_status(self, input_models, status):
-        # this might be called with the input which might be a filename or path
+    def _copy_models(self, input_models):
+        # Check whether input contains datamodels
+        make_copy = False
+        if isinstance(input_models, list):
+            is_datamodel = [isinstance(m, datamodels.JwstDataModel) for m in input_models]
+            if any(is_datamodel):
+                make_copy = True
+        elif isinstance(input_models, (datamodels.JwstDataModel, ModelContainer)):
+            make_copy = True
+
         if not isinstance(input_models, (datamodels.JwstDataModel, ModelLibrary, ModelContainer)):
+            # Input might be a filename or path.
+            # In that case, open it.
             input_models = datamodels.open(input_models)
```

Comment:
I find the fact that opening the datamodel is now hidden within a helper function labeled `_copy_models` confusing.  I wonder if this should be named `_open_models` instead, and in the docstring of the method state that it makes a copy if necessary to avoid modifying the inputs